src/function/built_in_functions.cpp

```diff
@@ -0,0 +1,749 @@
+    functions.insert({SIZE_FUNC_NAME, SizeFunction::getFunctionSet()});
+    functions.insert({LIST_EXTRACT_FUNC_NAME, ListExtractFunction::getFunctionSet()});
+    functions.insert({LIST_ELEMENT_FUNC_NAME, ListExtractFunction::getFunctionSet()});
+    functions.insert({LIST_CONCAT_FUNC_NAME, ListConcatFunction::getFunctionSet()});
+    functions.insert({LIST_CAT_FUNC_NAME, ListConcatFunction::getFunctionSet()});
+    functions.insert({ARRAY_CONCAT_FUNC_NAME, ListConcatFunction::getFunctionSet()});
+    functions.insert({ARRAY_CAT_FUNC_NAME, ListConcatFunction::getFunctionSet()});
+    functions.insert({LIST_APPEND_FUNC_NAME, ListAppendFunction::getFunctionSet()});
+    functions.insert({ARRAY_APPEND_FUNC_NAME, ListAppendFunction::getFunctionSet()});
+    functions.insert({ARRAY_PUSH_BACK_FUNC_NAME, ListAppendFunction::getFunctionSet()});
+    functions.insert({LIST_PREPEND_FUNC_NAME, ListPrependFunction::getFunctionSet()});
+    functions.insert({ARRAY_PREPEND_FUNC_NAME, ListPrependFunction::getFunctionSet()});
+    functions.insert({ARRAY_PUSH_FRONT_FUNC_NAME, ListPrependFunction::getFunctionSet()});
+    functions.insert({LIST_POSITION_FUNC_NAME, ListPositionFunction::getFunctionSet()});
+    functions.insert({ARRAY_POSITION_FUNC_NAME, ListPositionFunction::getFunctionSet()});
+    functions.insert({LIST_INDEXOF_FUNC_NAME, ListPositionFunction::getFunctionSet()});
+    functions.insert({ARRAY_INDEXOF_FUNC_NAME, ListPositionFunction::getFunctionSet()});
+    functions.insert({LIST_CONTAINS_FUNC_NAME, ListContainsFunction::getFunctionSet()});
+    functions.insert({LIST_HAS_FUNC_NAME, ListContainsFunction::getFunctionSet()});
+    functions.insert({ARRAY_CONTAINS_FUNC_NAME, ListContainsFunction::getFunctionSet()});
+    functions.insert({ARRAY_HAS_FUNC_NAME, ListContainsFunction::getFunctionSet()});
+    functions.insert({LIST_SLICE_FUNC_NAME, ListSliceFunction::getFunctionSet()});
+    functions.insert({ARRAY_SLICE_FUNC_NAME, ListSliceFunction::getFunctionSet()});
+    functions.insert({LIST_SORT_FUNC_NAME, ListSortFunction::getFunctionSet()});
+    functions.insert({LIST_REVERSE_SORT_FUNC_NAME, ListReverseSortFunction::getFunctionSet()});
+    functions.insert({LIST_SUM_FUNC_NAME, ListSumFunction::getFunctionSet()});
+    functions.insert({LIST_PRODUCT_FUNC_NAME, ListProductFunction::getFunctionSet()});
+    functions.insert({LIST_DISTINCT_FUNC_NAME, ListDistinctFunction::getFunctionSet()});
+    functions.insert({LIST_UNIQUE_FUNC_NAME, ListUniqueFunction::getFunctionSet()});
+    functions.insert({LIST_ANY_VALUE_FUNC_NAME, ListAnyValueFunction::getFunctionSet()});
+}
+
+void BuiltInFunctions::registerStructFunctions() {
+    functions.insert({STRUCT_PACK_FUNC_NAME, StructPackFunctions::getFunctionSet()});
+    functions.insert({STRUCT_EXTRACT_FUNC_NAME, StructExtractFunctions::getFunctionSet()});
+}
+
+void BuiltInFunctions::registerMapFunctions() {
+    functions.insert({MAP_CREATION_FUNC_NAME, MapCreationFunctions::getFunctionSet()});
```

Comment:
I start thinking maybe we should remove XXX_FUNC_NAME and just write them inside functions. Each function set will expose a getName interface. You can do this in a separate PR.